src/components/dialogs/free-properties-based-filter.js

```diff
@@ -0,0 +1,517 @@
+                    onChange={(oldVal, newVal) => {
+                        onChange(index, { name: newVal, values: [] });
+                        setName(newVal);
+                        setValues([]);
+                    }}
+                    options={predefinedNames}
+                    renderInput={(props) => (
+                        <TextField
+                            label={<FormattedMessage id="PropertyName" />}
+                            error={!!errors?.PropName}
+                            {...props}
+                        />
+                    )}
+                />
+            </FormControl>
+            <FormControl
+                sx={{ paddingLeft: '1ex', flexBasis: '72%', marginTop: 0 }}
+            >
+                <Autocomplete
+                    id="prop_values"
+                    value={values}
+                    freeSolo
+                    autoSelect
+                    forcePopupIcon
+                    multiple={true}
+                    onChange={(oldVal, newVal) => {
+                        onChange(index, { name, values: newVal });
+                        setValues(newVal);
+                    }}
+                    options={predefinedValues}
+                    renderInput={(props) => (
+                        <TextField
+                            label={<FormattedMessage id="PropertyValues" />}
+                            error={!!errors?.PropValue}
+                            {...props}
+                        />
+                    )}
+                    renderTags={(val, getTagsProps) =>
+                        val.map((code, index) => (
+                            <Chip
+                                id={'chip_' + code}
+                                size={'small'}
+                                label={code}
+                                {...getTagsProps({ index })}
+                            />
+                        ))
+                    }
+                />
+            </FormControl>
+        </>
+    );
+};
+
+export const FreeProperties = ({
+    initialValue,
+    onChange,
+    titleMessage,
+    validationsCount,
+    isForSubstation = false,
+}) => {
+    const numericSuffixRegex = /[0-9]*$/;
+    const numericSuffix = numericSuffixRegex.exec(titleMessage)[0];
+
+    const [fieldProps, setFieldProps] = useState(null);
+    const initialValues = useMemo(() => {
+        if (!initialValue) return [];
+        const ret = Object.entries(initialValue).map(([k, v]) => {
+            return { name: k || '', values: v };
+        });
+        return ret;
+    }, [initialValue]);
+    const validationsCountRef = useRef();
+
+    const onPropertiesArrayChange = useCallback(
+        (arr) => {
+            const obj = !arr
+                ? {}
+                : Object.fromEntries(arr.map((p) => [p.name || '', p.values]));
+            const vetoIdx = arr.findIndex((p) => !p.name || !p?.values?.length);
+            onChange(obj, vetoIdx >= 0);
```

Comment:
here is veto again ... a comment could help. What are we doing in this function ?